packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -0,0 +1,175 @@
+  Context = ExecutionContext,
+> = {
+  /**
+   * the worker's [bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/)
+   */
+  env: CloudflareEnv;
+  /**
+   * the request's [cf properties](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties)
+   */
+  cf: CfProperties | undefined;
+  /**
+   * the current [execution context](https://developers.cloudflare.com/workers/runtime-apis/context)
+   */
+  ctx: Context;
+};
+
+/**
+ * Symbol used as an index in the global scope to set and retrieve the Cloudflare context
+ *
+ * This is used both in production (in the actual built worker) and in development (`next dev`)
+ *
+ * Note: this symbol needs to be kept in sync with the one used in `src/cli/templates/worker.ts`
+ */
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
+ *
+ * @returns the cloudflare context
+ */
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(): CloudflareContext<CfProperties, Context> {
+  const global = getInternalGlobalThis<CfProperties, Context>();
+
+  const cloudflareContext = global[cloudflareContextSymbol];
+
+  if (!cloudflareContext) {
+    // the cloudflare context is initialized by the worker and is always present in production/preview
+    // during local development (`next dev`) it might be missing only if the developers hasn't called
+    // the `initOpenNextCloudflareForDev` function in their Next.js config file
+    const getContextFunctionName = getCloudflareContext.name;
+    const initFunctionName = initOpenNextCloudflareForDev.name;
+    throw new Error(
+      `\n\n\`${getContextFunctionName}\` has been called during development without having used` +
+        ` the \`${initFunctionName}\` function inside the Next.js config file.\n\n` +
+        `In order to use \`${getContextFunctionName}\` import and call ${initFunctionName} in the Next.js config file.\n\n` +
+        "Example: \n   ```\n   // next.config.mjs\n\n" +
+        `   import { ${initFunctionName} } from "@opennextjs/cloudflare";\n\n` +
+        `   ${initFunctionName}();\n\n` +
+        "   /** @type {import('next').NextConfig} */\n" +
+        "   const nextConfig = {};\n" +
+        "   export default nextConfig;\n" +
+        "   ```\n" +
+        "\n(note: currently middlewares in Next.js are always run using the edge runtime)\n\n"
+    );
+  }
+
+  return cloudflareContext;
+}
+
+/**
+ * Performs some initial setup to integrate as best as possible the local Next.js dev server (run via `next dev`)
+ * with the open-next Cloudflare adapter
+ *
+ * (Currently all the setup that this function performs is making sure that `getCloudflareContext` works as intended,
+ * in the future mode improvements might be added)
+ *
+ * Note: this function should only be called inside the Next.js config file
```

Comment:
add not that it doesn't need to be `await`ed?

Reply:
added 👍